src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/PubSubApiTest.cs

```diff
@@ -0,0 +1,237 @@
+                Assert.Equal(1, _messageCount);
+            }
+            finally
+            {
+                await ipfs.StopAsync();
+                cs.Cancel();
+            }
+        }
+
+        [Fact]
+        public async Task Subscribe_Mutiple_Messages()
+        {
+            _messageCount = 0;
+            var messages = "hello world this is pubsub".Split();
+            var topic = Guid.NewGuid().ToString();
+            var cs = new CancellationTokenSource();
+            await ipfs.StartAsync();
+            try
+            {
+                await ipfs.PubSubApi.SubscribeAsync(topic, msg => { Interlocked.Increment(ref _messageCount); }, cs.Token);
+                foreach (var msg in messages)
+                {
+                    await ipfs.PubSubApi.PublishAsync(topic, msg, cs.Token);
+                }
+
+                await Task.Delay(100, cs.Token);
+                Assert.Equal(messages.Length, _messageCount);
+            }
+            finally
+            {
+                await ipfs.StopAsync();
+                cs.Cancel();
+            }
+        }
+
+        [Fact]
+        public async Task Multiple_Subscribe_Mutiple_Messages()
+        {
+            _messageCount = 0;
+            var messages = "hello world this is pubsub".Split();
+            var topic = Guid.NewGuid().ToString();
+            var cs = new CancellationTokenSource();
+
+            void ProcessMessage(IPublishedMessage msg) { Interlocked.Increment(ref _messageCount); }
+            await ipfs.StartAsync();
+            try
+            {
+                await ipfs.PubSubApi.SubscribeAsync(topic, ProcessMessage, cs.Token);
+                await ipfs.PubSubApi.SubscribeAsync(topic, ProcessMessage, cs.Token);
+                foreach (var msg in messages)
+                {
+                    await ipfs.PubSubApi.PublishAsync(topic, msg, cs.Token);
+                }
+
+                await Task.Delay(100, cs.Token);
+                Assert.Equal(messages.Length * 2, _messageCount);
+            }
+            finally
+            {
+                await ipfs.StopAsync();
+                cs.Cancel();
+            }
+        }
+
+        private volatile int _messageCount1;
+
+        [Fact]
+        public async Task Unsubscribe()
+        {
+            _messageCount1 = 0;
+            var topic = Guid.NewGuid().ToString();
+            var cs = new CancellationTokenSource();
+            await ipfs.StartAsync();
+            try
+            {
+                await ipfs.PubSubApi.SubscribeAsync(topic, msg => { Interlocked.Increment(ref _messageCount1); }, cs.Token);
+                await ipfs.PubSubApi.PublishAsync(topic, "hello world!", default);
+                await Task.Delay(100, default);
+                Assert.Equal(1, _messageCount1);
+
+                cs.Cancel();
+                await ipfs.PubSubApi.PublishAsync(topic, "hello world!!!", default);
+                await Task.Delay(100, default);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)